@spotlightjs/overlay defines a `require` export with `umd` but the file is not generated

From https://github.com/getsentry/spotlight/issues/609#issuecomment-2539055113

Essentially, we defined a `require` field in `exports` inside `package.json`, pointing to a `spotlight.umd.js` file which doesn't exist since we don't build it.

I am not a very experienced node developer, so I am not fully certain what you are asking. But changing the `require` field of package.json to `sentry-spotlight.js` does indeed work.

I think our build system does not support `import`, because removing the `require` field causes the error `Module not found: Error: Package path . is not exported from package /tmp/broccoli-22414IdHLV4Arbl5L/cache-369-webpack_bundler_ember_auto_import_webpack/node_modules/@spotlightjs/overlay (see exports field in /tmp/broccoli-22414IdHLV4Arbl5L/cache-369-webpack_bundler_ember_auto_import_webpack/node_modules/@spotlightjs/overlay/package.json)`.

--- a/.changeset/heavy-monkeys-kneel.md
+++ b/.changeset/heavy-monkeys-kneel.md
@@ -1,0 +1,5 @@
+---
+'@spotlightjs/overlay': patch
+---
+
+Add missing UMD bundle

--- a/packages/overlay/vite.config.ts
+++ b/packages/overlay/vite.config.ts
@@ -3,6 +3,7 @@ import MagicString from 'magic-string';
 import { resolve, sep } from 'node:path';
 import type { PluginOption } from 'vite';
 import { defineConfig } from 'vite';
+import type { RenderedChunk } from 'rollup';
 import dts from 'vite-plugin-dts';
 import svgr from 'vite-plugin-svgr';
 
@@ -49,12 +50,17 @@ export default defineConfig({
       name: 'Spotlight',
       // the proper extensions will be added
       fileName: 'sentry-spotlight',
-      formats: ['es', 'iife'],
+      formats: ['es', 'iife', 'umd'],
     },
     rollupOptions: {
       treeshake: 'smallest',
       output: {
-        footer: '(function(S){S && S.init({integrations: [S.sentry(), S.console()]})}(window.Spotlight))',
+        footer(chunk: RenderedChunk) {
+          if (chunk.fileName.endsWith('.iife.js')) {
+            return '(function(S){S && S.init({integrations: [S.sentry(), S.console()]})}(window.Spotlight))';
+          }
+          return '';
+        },
       },
     },
     sourcemap: true,

